Python/ceval_macros.h

```diff
@@ -258,8 +324,21 @@ GETITEM(PyObject *v, Py_ssize_t i) {
 #define SETLOCAL(i, value)      do { _PyStackRef tmp = GETLOCAL(i); \
                                      GETLOCAL(i) = value; \
                                      PyStackRef_XCLOSE(tmp); } while (0)
-
-#define GO_TO_INSTRUCTION(op) goto PREDICT_ID(op)
+#ifdef Py_TAIL_CALL_INTERP
+#ifdef LLTRACE
+#define GO_TO_INSTRUCTION(op, size) do { \
+    Py_MUSTTAIL \
+    return (INSTRUCTION_TABLE[op])(frame, stack_pointer, tstate, next_instr - 1 - size, opcode, oparg, lltrace); \
+} while (0)
+#else
+#define GO_TO_INSTRUCTION(op, size) do { \
+    Py_MUSTTAIL \
+    return (INSTRUCTION_TABLE[op])(frame, stack_pointer, tstate, next_instr - 1 - size, opcode, oparg); \
+} while (0)
+#endif
+#else
+#define GO_TO_INSTRUCTION(op, size) goto PREDICT_ID(op)
```

Comment:
Can we do the transformations in the code generator?
Maybe drop the `GO_TO_INSTRUCTION` macro altogether and emit a plain `goto`?